File: frontend/src/lib/types/compliancePosture.ts
export type FrameworkRef = {
  framework: string;
  control_id: string;
};

export type ComplianceIssue = {
  id: string;
  title: string;
  severity: string;
  remediation: string;
  source: string;
  gap_type: string | null;
  control_ids: string[];
  framework_refs: FrameworkRef[];
  resource_type: string | null;
  resource_id: string | null;
};

export type FrameworkPosture = {
  id: string;
  name: string;
  description: string;
  status: string;
  alignment_score: number | null;
  summary: Record<string, number>;
  open_issue_count: number;
  open_issues: ComplianceIssue[];
  detail_url: string;
};

export type CompliancePosture = {
  evaluated_at: string;
  last_gap_analysis_at: string | null;
  frameworks: FrameworkPosture[];
  disclaimer: string;
};

export const FRAMEWORK_LABELS: Record<string, string> = {
  nist_ai_rmf: "NIST AI RMF",
  internal_guardrails: "Internal guardrails",
  gaira: "GAIRA governance",
};

export const POSTURE_STATUS_LABELS: Record<string, string> = {
  met: "Compliant",
  partial: "Partial",
  not_met: "Non-compliant",
  not_assessed: "Not assessed",
};

import type { StatusVariant } from "@/lib/utils";

export function postureStatusVariant(status: string): StatusVariant {
  switch (status) {
    case "met":
      return "success";
    case "partial":
      return "warning";
    case "not_met":
      return "danger";
    default:
      return "neutral";
  }
}

File: backend/app/schemas/compliance_posture.py
"""Unified compliance posture API schemas."""

from typing import Any

from pydantic import BaseModel, Field


class FrameworkRefResponse(BaseModel):
    framework: str
    control_id: str


class ComplianceIssueResponse(BaseModel):
    id: str
    title: str
    severity: str
    remediation: str
    source: str
    gap_type: str | None = None
    control_ids: list[str] = Field(default_factory=list)
    framework_refs: list[FrameworkRefResponse] = Field(default_factory=list)
    resource_type: str | None = None
    resource_id: str | None = None


class FrameworkPostureResponse(BaseModel):
    id: str
    name: str
    description: str
    status: str
    alignment_score: float | None = None
    summary: dict[str, Any] = Field(default_factory=dict)
    open_issue_count: int
    open_issues: list[ComplianceIssueResponse] = Field(default_factory=list)
    detail_url: str


class CompliancePostureResponse(BaseModel):
    evaluated_at: str
    last_gap_analysis_at: str | None = None
    frameworks: list[FrameworkPostureResponse]
    disclaimer: str
